### ISSUE TITLE
systemd spins on 618.0.0

### ISSUE THREAD

**Issue by crawford** [coreos/bugs#315]
Reproduced this on DigitalOcean.

```
$ cat /etc/systemd/system/etcd.service 
[Unit]
Description=etcd
After=docker.service
Requires=docker.service

[Service]
ExecStartPre=-/usr/bin/docker kill etcd
ExecStartPre=-/usr/bin/docker rm etcd
ExecStartPre=/usr/bin/docker pull quay.io/coreos/etcd:v0.4.8
ExecStart=/usr/bin/docker run -d -v /var/lib/etcd:/var/lib/etcd -p 4001:4001 -p 7001:7001 --name etcd quay.io/coreos/etcd:v0.4.8 \
          -name %m \ 
          -data-dir /var/lib/etcd \
          -addr 0.0.0.0:4001 \
          -peer-addr 0.0.0.0:7001 \
          -discovery https://discovery.etcd.io/32312f1ba66561284cca62ae0f2ad772
ExecStop=/usr/bin/docker stop etcd
Restart=always
RestartSec=10s
LimitNOFILE=40000
```

```
$ sudo systemctl start etcd
Failed to start etcd.service: Connection timed out
```

```
$ systemd-cgtop
Path                                                                            Tasks   %CPU   Memory  Input/s Output/s

/                                                                                  43   99.2    99.4M        -        -
/system.slice                                                                       -    1.1    86.6M        -        -
/system.slice/system-sshd.slice                                                     4    1.1    14.9M        -        -
/system.slice/ntpd.service                                                          1    0.0    11.8M        -        -
/system.slice/dbus.service                                                          1      -     1.0M        -        -
/system.slice/locksmithd.service                                                    1      -     6.6M        -        -
```

```
$ top
top - 19:07:20 up 7 min,  1 user,  load average: 1.00, 0.68, 0.31
Tasks:  58 total,   2 running,  56 sleeping,   0 stopped,   0 zombie
%Cpu(s): 76.9 us,  0.6 sy,  0.0 ni, 21.4 id,  0.1 wa,  0.8 hi,  0.1 si,  0.1 st
KiB Mem:    505876 total,   129856 used,   376020 free,     5120 buffers
KiB Swap:        0 total,        0 used,        0 free.    80216 cached Mem

  PID USER      PR  NI    VIRT    RES    SHR S %CPU %MEM     TIME+ COMMAND                                            
    1 root      20   0  110028   4952   3600 R 95.8  1.0   5:27.17 systemd
```

**Comment by crawford** [comment]
Unsurprisingly, this can be reproduced on 626.0.0.

**Comment by crawford** [comment]
Using this service file doesn't trigger the hang (removed the line continuation from ExecStart):

```
$ cat /etc/systemd/system/etcd.service 
[Unit]
Description=etcd
After=docker.service
Requires=docker.service

[Service]
ExecStartPre=-/usr/bin/docker kill etcd
ExecStartPre=-/usr/bin/docker rm etcd
ExecStartPre=/usr/bin/docker pull quay.io/coreos/etcd:v0.4.8
ExecStart=/usr/bin/docker run -d -v /var/lib/etcd:/var/lib/etcd -p 4001:4001 -p 7001:7001 --name etcd quay.io/coreos/etcd:v0.4.8 -name %m -data-dir /var/lib/etcd-addr 0.0.0.0:4001 -peer-addr 0.0.0.0:7001 -discovery https://discovery.etcd.io/32312f1ba66561284cca62ae0f2ad772
ExecStop=/usr/bin/docker stop etcd
Restart=always
RestartSec=10s
LimitNOFILE=40000
```

**Comment by crawford** [comment]
`systemctl status etcd` is enough to trigger the hang.

**Comment by crawford** [comment]
This is great. This file doesn't trigger the hang:

```
$ cat /etc/systemd/system/etcd.service 
[Unit]
Description=etcd
After=docker.service
Requires=docker.service

[Service]
ExecStartPre=-/usr/bin/docker kill etcd
ExecStartPre=-/usr/bin/docker rm etcd
ExecStartPre=/usr/bin/docker pull quay.io/coreos/etcd:v0.4.8
ExecStart=/usr/bin/docker run -d -v /var/lib/etcd:/var/lib/etcd -p 4001:4001 -p 7001:7001 --name etcd quay.io/coreos/etcd:v0.4.8 \
          -name %m \
          -data-dir /var/lib/etcd \
          -addr 0.0.0.0:4001 \
          -peer-addr 0.0.0.0:7001 \
          -discovery https://discovery.etcd.io/32312f1ba66561284cca62ae0f2ad772
ExecStop=/usr/bin/docker stop etcd
Restart=always
RestartSec=10s
LimitNOFILE=40000
```

What is the difference? "-name %m \" vs "-name %m \ " (trailing space).

**Comment by mischief** [comment]
this appears to have been fixed in systemd commit ba774317ac7d3e67fdb9ed81663264d38859df59
